### DISCUSSION TITLE
computeWindChillTemperature

### DISCUSSION THREAD

**Discussion by chunshek** [breezy-weather/breezy-weather#1136] [Ideas]
### Describe your idea

Half-bug, half new idea.

### Bug
`computeWindChillTemperature` calculates wind chill using a formula that takes km/h as parameter, but the function is currently passed `windSpeed` in m/s. As a result, all the calculated wind chills are incorrect.

### New idea
Environment Canada's [wind chill formula](https://climate.weather.gc.ca/glossary_e.html#w) itself has also been updated to cover two conditions:
1. Temperatures ≤ 0°C, wind speeds ≥ 5 km/h – existing formula
2. Temperatures ≤ 0°C, wind speeds < 5 km/h – simplified formula

[Previously](https://www.canada.ca/en/environment-climate-change/services/climate-change/canadian-centre-climate-services/display-download/technical-documentation-climate-normals.html) ECCC only used the first formula for temperatures ≤ 10°C and wind speeds ≥ 5 km/h. The NWS currently also only uses the first formula (albeit converted for imperial units), for temperatures ≤ 50°F (10°C) and wind speeds ≥ 3 mph (4.8 km/h).

I suggest updating `computeWindChillTemperature` to match ECCC's dual-condition formula, but keeping the threshold to 10°C (in line with previous Canadian and current US thresholds).

### Other details

Current ECCC formula for wind chill:
https://climate.weather.gc.ca/glossary_e.html#w

NWS formula for wind chill:
https://www.weather.gov/safety/cold-faqs

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.4 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by chunshek** [comment]
My modification is on [this branch](https://github.com/breezy-weather/breezy-weather/commit/fdea1f33e4e2cb1a4980d07a40383f4e281380e2).

**Reply by papjul** [reply]
Shouldn't we use the old formula for temperatures between 0 and 10 °C?

**Reply by chunshek** [reply]
We could. Your suggestion would mean:
| | T ≤ 0°C | 0°C < T ≤ 10°C | T > 10°C |
|-|-|-|-|
| WS ≥ 5 km/h | Old formula | Old formula | N/A |
| WS < 5 km/h | Simplified formula | N/A | N/A |

Whereas the code I have now would apply the simplified formula also for cases where 0°C < T ≤ 10°C and WS < 5 km/h.

Within this grey area, the discrepancy is at most −1.6°C when T = 0°C and WS = 5 km/h. The discrepancy is small enough I don't mind excluding this block from wind chill calculation. The **most important** thing is still to convert the wind speed from m/s to km/h before calculation. Let me know and I'll give it a quick fix.

**Reply by papjul** [reply]
LGTM

**Reply by chunshek** [reply]
Added a condition to only apply the simplified formula when T ≤ 0°C. [Here it is](https://github.com/breezy-weather/breezy-weather/commit/a783352b02eb0673682ca463ecdf2d29bfaf6427).

**Reply by papjul** [reply]
Merged, available in v5.2.5
